Src/IronPython/Runtime/ByteArray.cs

```diff
@@ -904,6 +905,20 @@ public class ByteArray : IList<byte>, ICodeFormattable, IBufferProtocol
             return new ByteArray(bytes);
         }
 
+        public static ByteArray operator +(ByteArray self, string other) {
+            if(other.Any(x => x > 0xff)) {
```

Comment:
`MakeByteArray` below does the same check as this. Maybe we could try catch the `MakeByteArray` and throw if necessary? Or if we don't want a try catch we could add a `TryMakeByteArray` which wouldn't throw.

Also, I wonder if we should add a note that this overload is specifically a workaround for the IPy string are unicode issue.